server/node/src/main/java/org/kaaproject/kaa/server/operations/service/akka/actors/core/ApplicationLogActorMessageProcessor.java

```diff
@@ -97,8 +97,12 @@ protected void processLogEventPack(ActorContext context, LogEventPackMessage mes
         List<LogAppender> required = filterAppenders(logSchema.getVersion(), true);
         List<LogAppender> optional = filterAppenders(logSchema.getVersion(), false);
         if (required.size() > 0 || optional.size() > 0) {
-            for (LogAppender logAppender : optional) {
-                logAppender.doAppend(message.getLogEventPack(), voidCallback);
+            if (required.size() > 0) {
+                for (LogAppender logAppender : optional) {
+                    logAppender.doAppend(message.getLogEventPack(), voidCallback);
+                }
+            } else {
+                required = optional;
             }
             LogDeliveryCallback callback;
             if (required.size() > 1) {
```

Comment:
We create different instances of LogDeliveryCallback here. In case required.size() == 0 we need to immediately execute same logic that is present in sendSuccessToendpoint method: https://github.com/kaaproject/kaa/blob/feature/KAA-832/server/node/src/main/java/org/kaaproject/kaa/server/operations/service/akka/messages/core/logs/AbstractActorCallback.java#L22